lib/options-handler.ts

```diff
@@ -397,9 +393,9 @@ export class ClientOptionsHandler {
     }
 
     async getRemoteLibraries(language: LanguageKey, remoteUrl: string) {
-        const remoteId = this.getRemoteId(remoteUrl, language);
+        const remoteId = getRemoteId(remoteUrl, language);
         if (!this.remoteLibs[remoteId]) {
-            return new Promise(resolve => {
+            return await new Promise(resolve => {
```

Comment:
I added await here to avoid duplicate work, as `this.remoteLibs[remoteId]` is not directly populated